File: OMGStarWars/app/src/main/java/dev/radley/omgstarwars/models/Starship.java
package dev.radley.omgstarwars.models;

import com.google.gson.annotations.SerializedName;

import java.io.Serializable;

/**
 * Starship model represents a single transport craft that has hyperdrive capability.
 */
public class Starship extends Vehicle implements Serializable {

    @SerializedName("starship_class")
    public String starshipClass;

    @SerializedName("hyperdrive_rating")
    public String hyperdriveRating;

    @SerializedName("MGLT")
    public String mglt;
}

File: OMGStarWars/app/src/main/java/dev/radley/omgstarwars/models/Starship.kt
package dev.radley.omgstarwars.models

import com.google.gson.annotations.SerializedName

import java.io.Serializable

/**
 * Starship model represents a single transport craft that has hyperdrive capability.
 */
class Starship : Vehicle(), Serializable {

    @SerializedName("starship_class")
    var starshipClass: String = ""

    @SerializedName("hyperdrive_rating")
    var hyperdriveRating: String = ""

    @SerializedName("MGLT")
    var mglt: String = ""
}
